### PR TITLE
[MRG] Fix bf proj with bads

### PR DESCRIPTION
**Pull request by agramfort**
make_lcmv and make_dics were computing filters on full meg data even in the presence of bads which caused a crash when applied to data with bad channels as proj matrix was suddenly different.

cc @britta-wstnr @wmvanvliet

### CODE REVIEW

File: mne/beamformer/tests/test_lcmv.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def _get_data(tmin=-0.1, tmax=0.15, all_forward=True, epochs=True,
     picks = mne.pick_types(raw.info, selection=left_temporal_channels)
     picks = picks[::2]  # decimate for speed
     raw.pick_channels([raw.ch_names[ii] for ii in picks])
+    raw.info['bads'] = raw.ch_names[:10]  # add more bads
```
Comment (larsoner) [inline mne/beamformer/tests/test_lcmv.py]: Did you test that things still pass locally? IIRC the LCMV/DICS tests were a bit sensitive to changing some of these params

Comment (agramfort) [reply mne/beamformer/tests/test_lcmv.py]: it runs for me locally with this change. but it would fail before.

**Comment by larsoner** [comment]
Let's hear from @britta-wstnr and @wmvanvliet in case they can look, if not let's merge at the end of the week since it's pretty straightforward

File: mne/beamformer/tests/test_lcmv.py (hunk 1 above)
Comment (britta-wstnr) [reply mne/beamformer/tests/test_lcmv.py]: seems 10 bads were a bit too much for the beamformer output, it moved out of the boundaries (``AssertionError``) for testing the normal orientation.

Comment (britta-wstnr) [reply mne/beamformer/tests/test_lcmv.py]: Furthermore, for completion: in ``test_lcmv_source_power``, ``picks`` has larger values than the number of remaining sensors.

**Comment by britta-wstnr** [comment]
Except for the tests, I am fine with the changes - I was there when they got implemented :smile:, so good to go from my side when tests fixed.

**Comment by larsoner** [comment]
> it runs for me locally with this change. but it would fail before.

The test you had didn't seem to make it fail for me. Maybe I did it wrong / didn't revert the fix properly or something. My modification didn't seem to make it fail, either. Not sure what the proper test would be.

In the meantime, @britta-wstnr can you look to make sure my commit makes sense?

**Comment by agramfort** [comment]
thx heaps @larsoner for finishing this one.

**Comment by larsoner** [comment]
backported

**Pull request merged**